src/loader.ts

```diff
@@ -11,19 +11,33 @@ import { DotenvOptions, setupDotenv } from "./dotenv";
 
 export type UserInputConfig = Record<string, any>;
 
+export interface ConfigLayerMeta {
+  name?: string;
+  [key: string]: any;
+}
+
 export interface C12InputConfig {
   extends?: string | string[];
   $envName?: string;
   $test?: UserInputConfig;
   $development?: UserInputConfig;
   $production?: UserInputConfig;
   $env?: Record<string, UserInputConfig>;
+  $meta?: ConfigLayerMeta;
 }
 
 export interface InputConfig extends C12InputConfig, UserInputConfig {}
 
+export interface SourceOptions {
+  meta?: ConfigLayerMeta;
```

Comment:
Don't think it make sense here

Reply:
https://github.com/unjs/c12/pull/62#discussion_r1133725764